### PR TITLE
Release notes for v1.0-rc.2

### PR DESCRIPTION
**Pull request by bdarnell**
As of cockroachdb/cockroach@198ba3aa75a0802b46aa2bea99540aa0d9004222

I'll be out tomorrow so here's an early version of the release notes so far. Update this with whatever else makes the cut.

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/1357)

### CODE REVIEW

File: v1.0-rc.2.md [hunk 1]
```diff
@@ -50,13 +50,16 @@ Get future release notes emailed to you:
...
+- Add tooltips for graphs on overview page. [#15624](https://github.com/cockroachdb/cockroach/pull/15624)
 
 ### Bug Fixes
 
 - Fixed a panic when a query returning a constant `NULL` used the new distributed execution engine. [#15530](https://github.com/cockroachdb/cockroach/pull/15530)
 - Fixed two issues that could cause requests to get stuck indefinitely. [#15557](https://github.com/cockroachdb/cockroach/pull/15557) [#15573](https://github.com/cockroachdb/cockroach/pull/15573)
 - Fixed a rare crash when extending table leases. [#15560](https://github.com/cockroachdb/cockroach/pull/15560)
 - Fixed a crash when a transaction that uses `SAVEPOINT cockroach_restart` does something other than `ROLLBACK TO SAVEPOINT` after an error. [#15563](https://github.com/cockroachdb/cockroach/pull/15563)
+- Fixed `RESTORE` for interleaved tables. [#15564](https://github.com/cockroachdb/cockroach/pull/15564)
+- Return parse error instead of panic when constant typechecking fails. [#15636](https://github.com/cockroachdb/cockroach/pull/15636)
```
Comment (andreimatei) [inline v1.0-rc.2.md]: + Backfills caused by `ALTER TABLE ADD COLUMN .. DEFAULT unique_rowid()` don't generate duplicates anymore. [#15628](https://github.com/cockroachdb/cockroach/pull/15628)

File: v1.0-rc.2.md [hunk 2]
```diff
@@ -50,13 +50,16 @@ Get future release notes emailed to you:
...
 
 ### Performance Improvements
```
Comment (andreimatei) [inline v1.0-rc.2.md]: + Queries using session-dependent variables (e.g. time conversions using the session's timezone) are now executed through the Distributed SQL engine, whereas before they were blacklisted.

File: v1.0-rc.2.md (hunk 1 above)
Comment (petermattis) [inline v1.0-rc.2.md]: `Backfilling` is an implementation detail. How about: Adding a new column with a `DEFAULT unique_rowid()` specification no longer generates duplicate values.

Comment (andreimatei) [inline v1.0-rc.2.md]: well if the table is empty (or, if the column is part of the original CREATE TABLE), that was fine before too, so I said something about backfilling to not scare people too much.

File: v1.0-rc.2.md (hunk 2 above)
Comment (mberhault) [inline v1.0-rc.2.md]: do you have a PR for that?

**Comment by andreimatei** [comment]
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[v1.0-rc.2.md, line 65 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/1357#-KjB5iTdKcrnlC0FBcI3-r2-65:-KjEdmafBQGVutRrLTO6:b-w68js5) ([raw file](https://github.com/cockroachdb/docs/blob/4869437b3c315f1367965f38c4976677663ffa53/v1.0-rc.2.md#L65)):*
> [#15636](https://github.com/cockroachdb/cockroach/pull/15636)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1357)*

File: v1.0-rc.2.md (hunk 1 above)
Comment (petermattis) [inline v1.0-rc.2.md]: Adding a new column via `ALTER TABLE ADD COLUMN` with a `DEFAULT unique_rowid()` specification no longer generates duplicate values.

**Comment by mberhault** [comment]
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[v1.0-rc.2.md, line 65 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/1357#-KjB5iTdKcrnlC0FBcI3-r2-65:-KjEe0z2vlU4VyOOdyeQ:b-4hmeo2) ([raw file](https://github.com/cockroachdb/docs/blob/4869437b3c315f1367965f38c4976677663ffa53/v1.0-rc.2.md#L65)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

> [#15636](https://github.com/cockroachdb/cockroach/pull/15636)
</blockquote></details>

uh. no, that's for `sql: fix panic in constant typechecking`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1357)*

**Comment by andreimatei** [comment]
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[v1.0-rc.2.md, line 62 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/1357#-KjB5iTdKcrnlC0FBcI3-r2-62:-KjEeFa1vragsP2JDEnF:b7hy0nt) ([raw file](https://github.com/cockroachdb/docs/blob/4869437b3c315f1367965f38c4976677663ffa53/v1.0-rc.2.md#L62)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Adding a new column via `ALTER TABLE ADD COLUMN` with a `DEFAULT unique_rowid()` specification no longer generates duplicate values.
</blockquote></details>

thumbs up

---

*[v1.0-rc.2.md, line 65 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/1357#-KjB5iTdKcrnlC0FBcI3-r2-65:-KjEeP1aI9bBzNrObiU_:byv4hvl) ([raw file](https://github.com/cockroachdb/docs/blob/4869437b3c315f1367965f38c4976677663ffa53/v1.0-rc.2.md#L65)):*
<details><summary><i>Previously, mberhault (marc) wrote…</i></summary><blockquote>

uh. no, that's for `sql: fix panic in constant typechecking`
</blockquote></details>

sorry ,meant [#15368](https://github.com/cockroachdb/cockroach/pull/15368)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1357)*

**Comment by mberhault** [comment]
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[v1.0-rc.2.md, line 65 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/1357#-KjB5iTdKcrnlC0FBcI3-r2-65:-KjEf4dqnMmWGdgV_0Ko:bq62bw) ([raw file](https://github.com/cockroachdb/docs/blob/4869437b3c315f1367965f38c4976677663ffa53/v1.0-rc.2.md#L65)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary><blockquote>

sorry ,meant [#15368](https://github.com/cockroachdb/cockroach/pull/15368)
</blockquote></details>

you sure? that says `distsql: avoid unhealthy nodes during planning` which I've put in the release notes.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1357)*

**Comment by andreimatei** [comment]
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[v1.0-rc.2.md, line 65 at r2](https://reviewable.io:443/reviews/cockroachdb/docs/1357#-KjB5iTdKcrnlC0FBcI3-r2-65:-KjEgdsqaCrc6jbbEkXw:btim1z1) ([raw file](https://github.com/cockroachdb/docs/blob/4869437b3c315f1367965f38c4976677663ffa53/v1.0-rc.2.md#L65)):*
<details><summary><i>Previously, mberhault (marc) wrote…</i></summary><blockquote>

you sure? that says `distsql: avoid unhealthy nodes during planning` which I've put in the release notes.
</blockquote></details>

It's #15628. I'm sorry...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1357)*

**Comment by andreimatei** [comment]
LGTM

**Comment by jseldess** [comment]
*[v1.0-rc.2.md, line 33 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1357#-KjEo_GrHeI_gqKQnJbQ:-KjEo_Gs3NP2QnsXT7cx:bk7zo87) ([raw file](https://github.com/cockroachdb/docs/blob/620f555eaa27572c9fe2222c51ba94fc2ea9a135/v1.0-rc.2.md#L33)):*
> ```Markdown
> ### General Changes
> 
> - [Diagnostics reporting](diagnostics-reporting.html) now anonymized query statistics. [#15491](https://github.com/cockroachdb/cockroach/pull/15491)
> ```

`now includes anonymized query statistics`. I'll add this myself, along with updated diagnostics docs.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1357)*

**Comment by mberhault** [comment]
Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions.

---

*[v1.0-rc.2.md, line 33 at r3](https://reviewable.io:443/reviews/cockroachdb/docs/1357#-KjEo_GrHeI_gqKQnJbQ:-KjEvgUKijZVhCYw5BjL:bb6un1j) ([raw file](https://github.com/cockroachdb/docs/blob/620f555eaa27572c9fe2222c51ba94fc2ea9a135/v1.0-rc.2.md#L33)):*
<details><summary><i>Previously, jseldess wrote…</i></summary><blockquote>

`now includes anonymized query statistics`. I'll add this myself, along with updated diagnostics docs.
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1357)*

**Comment by mberhault** [comment]
I added a `source` link. However, we don't really have a button for it.

**Comment by jseldess** [comment]
@mberhault, the default button looks just fine for now. I've updated the CSS to handle 4 buttons. 

Is it easy to track down the source url for all previous release as well? Might be nice to add the button to the other release note pages.

**Comment by mberhault** [comment]
Yup. we have the following source files built:
```
$ aws s3 ls s3://binaries.cockroachdb.com|grep src
2017-04-07 10:57:25   59494344 cockroach-beta-20170330.src.tgz
2017-04-07 11:33:37   60021159 cockroach-beta-20170406.src.tgz
2017-04-13 17:18:28   61096138 cockroach-beta-20170413.src.tgz
2017-04-20 16:29:15   68343752 cockroach-beta-20170420.src.tgz
2017-05-01 14:30:19   68754332 cockroach-latest.src.tgz
2017-05-01 14:30:04   68784067 cockroach-v1.0-rc.1.src.tgz
```

So back to (and including) `beta-20170330`

**Comment by mberhault** [comment]
Nice! thank you.
I'll add a few more things once we're sorted out the cherrypicking.

File: v1.0-rc.2.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+- The distributed SQL engine supports queries that internally generate rows with no columns. [#15420](https://github.com/cockroachdb/cockroach/pull/15420)
+- New variables added to the [cluster-wide settings](cluster-settings.html): `server.declined_reservation_timeout`, `server.failed_reservation_timeout`, and `server.time_until_store_dead`. [#15581](https://github.com/cockroachdb/cockroach/pull/15581)
+
+### Command-Line Interface Changes
+
+- If the new flag `--listening-url-file` is passed to [`cockroach start`](start-a-node.html), the server will write its SQL connectuion URL to the specified file on successful startup, in addition to printing it to `stdout`. This is particularly helpful in identifying the node's port when an unused port is assigned automatically (`--port=0`).[#15468](https://github.com/cockroachdb/cockroach/pull/15468)
```
Comment (spencerkimball) [inline v1.0-rc.2.md]: s/connectuion/connection/

Comment (jseldess) [inline v1.0-rc.2.md]: Done.

File: v1.0-rc.2.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+- The [`cockroach sql`](use-the-built-in-sql-client.html) shell no longer buffers all results in memory before printing them. [#15277](https://github.com/cockroachdb/cockroach/pull/15277)
+- The log file is now flushed more reliably when a node exits. [#15531](https://github.com/cockroachdb/cockroach/pull/15531)
+- The new flag `--max-offset` on the [`cockroach start`](start-a-node.html) command defines the maximum allowed clock offset for the cluster and replaces the environment variable `COCKROACH_MAX_OFFSET`. Note that the `--max-offset` value must be the same on all nodes in the cluster. In order to change it, every node in the cluster must be stopped and restarted with the new value. [#15568](https://github.com/cockroachdb/cockroach/pull/15568)
+- The `--raft-tick-interval` flag has been removed from the [`cockroach start`](start-a-node.html) command. [#15547](https://github.com/cockroachdb/cockroach/pull/15547)
+- Surface errors encountered during certificate and key load. [#15655](https://github.com/cockroachdb/cockroach/pull/15655)
```
Comment (knz) [inline v1.0-rc.2.md]: Flip the sentence: "Errors encountered during certificate and key load are now reported to the user in the commands where they generate."

File: v1.0-rc.2.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+### Admin UI Changes
+
+- The number of transactions that would be affected by a change in isolation level is now tracked. [#15234](https://github.com/cockroachdb/cockroach/pull/15234)
+- Improved style of the single node and log pages. [#15486](https://github.com/cockroachdb/cockroach/pull/15486)
+- The event log now refreshes automatically. [#15502](https://github.com/cockroachdb/cockroach/pull/15502)
+- Add tooltips for graphs on overview page. [#15624](https://github.com/cockroachdb/cockroach/pull/15624)
```
Comment (knz) [inline v1.0-rc.2.md]: Graphs on the overview page now have tooltips.

File: v1.0-rc.2.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+- Fixed two issues that could cause requests to get stuck indefinitely. [#15557](https://github.com/cockroachdb/cockroach/pull/15557) [#15573](https://github.com/cockroachdb/cockroach/pull/15573)
+- Fixed a rare crash when extending table leases. [#15560](https://github.com/cockroachdb/cockroach/pull/15560)
+- Fixed a crash when a transaction that uses [`SAVEPOINT cockroach_restart`](savepoint.html) does something other than `ROLLBACK TO SAVEPOINT` after an error. [#15563](https://github.com/cockroachdb/cockroach/pull/15563)
+- Fixed [`RESTORE`](restore.html) for interleaved tables. [#15564](https://github.com/cockroachdb/cockroach/pull/15564)
+- Return parse error instead of panic when constant typechecking fails. [#15636](https://github.com/cockroachdb/cockroach/pull/15636)
+- Prevent persistent distsql and schema change errors after a node dies. [#15368](https://github.com/cockroachdb/cockroach/pull/15368)
+- Adding a new column via [`ALTER TABLE ADD COLUMN`](add-column.html) with a `DEFAULT unique_rowid()` specification no longer generates duplicate values. [#15628](https://github.com/cockroachdb/cockroach/pull/15628)
```
Comment (knz) [inline v1.0-rc.2.md]: "no longer generates" -> "much more rarely generates"
(they still may generated if a node's clock jumps backwards during the back fill)

Comment (knz) [inline v1.0-rc.2.md]: https://github.com/cockroachdb/docs/issues/1373

Comment (jseldess) [inline v1.0-rc.2.md]: Done.

File: v1.0-rc.2.md (hunk 5 above)
Comment (jseldess) [inline v1.0-rc.2.md]: Done.

File: v1.0-rc.2.md (hunk 4 above)
Comment (jseldess) [inline v1.0-rc.2.md]: Done.

File: v1.0-rc.2.md [hunk 7]
```diff
@@ -64,7 +64,7 @@ Get future release notes emailed to you:
 - Fixed [`RESTORE`](restore.html) for interleaved tables. [#15564](https://github.com/cockroachdb/cockroach/pull/15564)
 - Return parse error instead of panic when constant typechecking fails. [#15636](https://github.com/cockroachdb/cockroach/pull/15636)
 - Prevent persistent distsql and schema change errors after a node dies. [#15368](https://github.com/cockroachdb/cockroach/pull/15368)
-- Adding a new column via [`ALTER TABLE ADD COLUMN`](add-column.html) with a `DEFAULT unique_rowid()` specification no longer generates duplicate values. [#15628](https://github.com/cockroachdb/cockroach/pull/15628)
+- Adding a new column via [`ALTER TABLE ADD COLUMN`](add-column.html) with a `DEFAULT unique_rowid()` specification no longer generates duplicate values, except in cases where a node's clock jumps backwards during the back fill. [#15628](https://github.com/cockroachdb/cockroach/pull/15628)
```
Comment (knz) [inline v1.0-rc.2.md]: No this was a mistake in my previous comment. There is a protection in the code against clock jumps, even when that happens the rowids will still be different. The conditions for duplicates are very subtle, I'm not sure we can make a good summary in these release notes.

Comment (jseldess) [inline v1.0-rc.2.md]: OK, changed back to "very rarely generates..."

**Comment by andreimatei** [comment]
Review status: 0 of 10 files reviewed at latest revision, 8 unresolved discussions.

---

*[v1.0-rc.2.md, line 67 at r11](https://reviewable.io:443/reviews/cockroachdb/docs/1357#-KjB5iTdKcrnlC0FBcI3-r11-67:-KjNvmsX68fNtgTImRsj:b9ni9gq) ([raw file](https://github.com/cockroachdb/docs/blob/c9e124dfb5f968d8ab1fff088914c70792c2e7e7/v1.0-rc.2.md#L67)):*
<details><summary><i>Previously, jseldess wrote…</i></summary><blockquote>

OK, changed back to "very rarely generates..."
</blockquote></details>

IMHO, this sounds alarmist without good reason. Rafa, what do you have in mind exactly? A node restarting, and the clock jumping back more than MAX_OFFSET at some point (we sleep for MAX_OFFSET on startup)?
I wouldn't say anything in these release notes...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/docs/1357)*

**Comment by knz** [comment]
@andreimatei check out the issue I filed about this #1373

**Pull request merged**